addons/mail/static/src/views/fields/badge_selection_icons/badge_selection_icons_field.js

```diff
@@ -47,7 +47,10 @@ export class BadgeSelectionWithIconsField extends BadgeSelectionField {
                 });
 
                 return ret.map((opt) => {
-                    const iconValue = opt[props.relatedIconField] || props.defaultIcon;
+                    let iconValue = opt[props.relatedIconField] || props.defaultIcon;
+                    if (props.iconMapping) {
+                        iconValue = props.iconMapping[iconValue] || props.defaultIcon;
```

Comment:
A bit strange to read value for defaultIcon in iconMapping.

Maybe this:
```js
const relatedFieldValue = opt[props.relatedIconField];
// In case the icon value is a selection, we can specify a mapping too
const iconValue = props.iconMapping ? props.iconMapping[relatedFieldValue] : relatedFieldValue;
return [opt.id, opt.name, iconValue || props.defaultIcon];
```